custom_metrics/wpt_bodies.js

```diff
@@ -0,0 +1,1366 @@
+        }
+
+        return result; 
+      }
+      catch(e) {
+        return logError("meta_description", e);
+      }
+    })(),
+
+    // Extract hreflang info  
+    // https://support.google.com/webmasters/answer/189077?hl=en
+    // Used by: SEO
+    // Backup: link nodes in almanac.js 
+    'hreflangs': (() => {
+      try {    
+        let result = {http_header: {values: []}};
+
+        function getHreflangValues(d) {
+          let target = {values: []};
+          let hreflangs = Array.from(d.querySelectorAll('link[rel="alternate"][hreflang]'));
+
+          hreflangs.forEach(e => {
+            target.values.push(e.getAttribute("hreflang"));
+          });
+
+          return target;
+        }
+
+        result.rendered = getHreflangValues(document);
+
+        let rawHtmlDocument = getRawHtmlDocument();
+
+        if (rawHtmlDocument) {
+          result.raw = getHreflangValues(rawHtmlDocument);
+        }
+
+        let linkHeaders = getResponseHeaders("link");
+        if (linkHeaders) {
+          linkHeaders.forEach((h) => {
+            let matches = h.matchAll(/hreflang=['"]?(.*?)['"]/gi);
+
+            for (const match of matches) {
+              let c =match[1];
+              result.http_header.values.push(c);
+            }  
+          })
+        }
+
+        return result; 
+      }
+      catch(e) {
+        return logError("hreflangs", e);
+      }
+    })(),
+
+    // heading information from H1 to H8 
+    // Used by: SEO, Markup
+    // Backup: heading order and length of h1s in almanac.js - not much
+    'headings': (() => { 
+      try {  
+      
+        function processHeading(d, target, n, level) {
+          let html  = n.innerHTML;
+          let text = seoText(n);
+
+          let words = text.match(/\S+/g)?.length;
+
+          if (!target.primary) {
+            if (text.length > 0) { // make this primary
+              let primaryTitle = "";
+              if (d == document) {
+                primaryTitle = renderedPrimaryTitle; 
+              }
+              else {
+                primaryTitle = rawPrimaryTitle; 
+              }
+
+              let snippet = text;
+
+              if (snippet.length > 100) {
+                snippet = snippet.substring(0, 100)+"...";
+              }
+
+              target.primary = {
+                words: words,
+                characters: text.length,
+                matches_title: text.toLowerCase() == primaryTitle?.toLowerCase(),
+                text: snippet,
+                level: level
+              }
+            } else if (html.length > 0) {
+              // looks like a hidden heading
+              target.first_non_empty_heading_hidden = true;
+            }
+          }
+          return {characters: text.length, words: words ?? 0};
+        }
+        
+        function processHeadings(d) {
+          let target = {first_non_empty_heading_hidden: false};
+          for(let l=1; l < 9; l++) {
```

Comment:
H6 is the lowest level per https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements. Why go to 8?

Reply:
🤷  ok